expression/builtin_like_vec.go

```diff
@@ -27,17 +30,80 @@ func (b *builtinLikeSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) er
 }
 
 func (b *builtinRegexpBinarySig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinRegexpBinarySig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	return vecEvalRegexp(&b.baseBuiltinFunc, b, input, result)
 }
 
 func (b *builtinRegexpSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinRegexpSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	return vecEvalRegexp(&b.baseBuiltinFunc, b, input, result)
+}
+
+func vecEvalRegexp(b *baseBuiltinFunc, rc regexpCompiler, input *chunk.Chunk, result *chunk.Column) error {
+	n := input.NumRows()
+	bufExpr, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(bufExpr)
+	if err := b.args[0].VecEvalString(b.ctx, input, bufExpr); err != nil {
+		return err
+	}
+
+	bufPat, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(bufPat)
+	if err := b.args[1].VecEvalString(b.ctx, input, bufPat); err != nil {
+		return err
+	}
+
+	pat2re := make(map[string]*regexp.Regexp)
+	isMemoizable := b.args[1].ConstItem()
+	if isMemoizable {
+		// If args[1] is a constant item, then all the items in bufPat are
+		// the same. So the pattern can be compiled once and memoized.
+		for i := 0; i < n; i++ {
+			if bufPat.IsNull(i) {
+				continue
+			}
+			pat := bufPat.GetString(i)
+			if _, ok := pat2re[pat]; ok {
+				continue
+			}
+			re, err := rc.compile(pat)
+			if err != nil {
+				return err
+			}
+			pat2re[pat] = re
+		}
+	}
```

Comment:
What about compiling once and do some special check?

Reply:
Ack, PTAL.